examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -0,0 +1,318 @@
+          512. 1024. 1536. 2048. 2560. 3072. 3584. 4096. 4608. 5120. 5632. 6144. 6656. 7168. 7680. 8192.
+
+DECLARE_SHADER array_to_blocked_array KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding = 0) buffer array1 {
+  float[MAT_SIZE][MAT_SIZE] data;
+};
+layout(std430, binding=1) buffer array1_mat_buf{
+  mat4x4 array_mat[4][4];
+};
+void main() {
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
+    array_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data[row][col];
+}
+END
+
+COMPILE_SHADER shader_compiled_1 SHADER array_to_blocked_array
+CREATE_PROGRAM compute_prog_1 SHADERS shader_compiled_1
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_1
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array1_mat_buf correct_array1_mat_buf
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array2 BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+
+RUN_COMPUTE
+    PROGRAM compute_prog_1
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS array2_mat_buf correct_array2_mat_buf
+
+DECLARE_SHADER mat_mul KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding=0) buffer array1_mat_buf{
+  mat4x4 array1_mat[4][4];
+};
+layout(std430, binding=1) buffer array2_mat_buf{
+  mat4x4 array2_mat[4][4];
+};
+layout(std430, binding=2) buffer blocked_result{
+  mat4x4 blocked_res[4][4];
+};
+
+void main() {
+    for(int i = 0;i < MAT_SIZE/KERNEL_SIZE; i++){
+        for(int j = 0; j < MAT_SIZE/KERNEL_SIZE; j++){
+            blocked_res[i][j] = mat4x4(0);
+            for(int k = 0; k < MAT_SIZE/KERNEL_SIZE; k++){
+                 // The submatrix at position (i,j) is equal to the multiplication
+                // of all the submatrices (i,k) and (k,j) where k = 0....(MAT_SIZE/KERNEL_SIZE).
+                blocked_res[i][j] += array1_mat[i][k] * array2_mat[k][j];
+            }
+        }
+    }
+}
+END
+
+COMPILE_SHADER shader_compiled_3 SHADER mat_mul
+CREATE_PROGRAM compute_prog_3 SHADERS shader_compiled_3
+
+BIND_SHADER_STORAGE_BUFFER BUFFER array1_mat_buf BINDING 0
+BIND_SHADER_STORAGE_BUFFER BUFFER array2_mat_buf BINDING 1
+BIND_SHADER_STORAGE_BUFFER BUFFER blocked_result BINDING 2
+
+RUN_COMPUTE
+    PROGRAM compute_prog_3
+    NUM_GROUPS 16 16 1
+
+ASSERT_EQUAL BUFFERS blocked_result correct_blocked_result
+
+DECLARE_SHADER blocked_array_to_array KIND COMPUTE
+#version 450
+#define KERNEL_SIZE 4
+#define MAT_SIZE  16
+
+layout(local_size_x=1, local_size_y=1, local_size_z=1) in;
+
+layout(std430, binding=0) buffer result{
+  float[MAT_SIZE][MAT_SIZE] res;
+};
+layout(std430, binding=1) buffer blocked_result{
+  mat4x4 blocked_res[4][4];
+};
```

Comment:
Similar comments - change the occurrences of 4, and move array sizes to after res.